api/types.proto

```diff
@@ -696,3 +696,9 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+message Device {
+	string source = 1;
+	string target = 2;
+	string permissions = 3;
```

Comment:
So, it sounds like we also need to be able to parameterize this over some sort of device class, then. Let's say that we have something on `NodeSpec`, configured by a cluster operator:

```
NodeSpec{
  Devices: [
   {Class: "/dev/nvidia", Available: ["/dev/nvidia0", "/dev/nvidia1"]}
  ],
}
```

This could also be announced through `NodeDescription`, then cluster operators could configure the ACL (access+class matching) for devices, such that these would provide control over which devices are mappable in the container configuration. Ultimately availability would then be reported through `Node`, as a product of `NodeDescription` and the current ACL. 

Two tasks could theoretically be dispatched, one using `/dev/nvidia0` and the other `/dev/nvidia1`, with only the `/dev/nvidia` definition present in the `ContainerSpec`. Containers wouldn't necessarily get `/dev/nvidia0`, but they would get something that provides it.